src/uct/ib/rc/base/rc_ep.c

```diff
@@ -374,42 +382,60 @@ ucs_arbiter_cb_result_t uct_rc_ep_process_pending(ucs_arbiter_t *arbiter,
     return UCS_ARBITER_CB_RESULT_DESCHED_GROUP;
 }
 
+ucs_arbiter_cb_result_t
+uct_rc_ep_arbiter_purge_internal_cb(ucs_arbiter_t *arbiter,
+                                    ucs_arbiter_group_t *group,
+                                    ucs_arbiter_elem_t *elem, void *arg)
+{
+    uct_pending_req_t *req = ucs_container_of(elem, uct_pending_req_t, priv);
+    uct_rc_ep_t *ep        = ucs_container_of(group, uct_rc_ep_t, arb_group);
+    uct_rc_pending_req_t *freq;
+
+    if (req->func == uct_rc_ep_check_progress) {
+        ep->flags &= ~UCT_RC_EP_FLAG_KEEPALIVE_PENDING;
+        ucs_mpool_put(req);
+    } else if (req->func == uct_rc_ep_fc_grant) {
+        freq = ucs_derived_of(req, uct_rc_pending_req_t);
+        ucs_mpool_put(freq);
+    } else {
+        /* Non-internal request was found */
+        return UCS_ARBITER_CB_RESULT_RESCHED_GROUP;
+    }
+
+    /* Internal request was found */
+    return UCS_ARBITER_CB_RESULT_REMOVE_ELEM;
```

Comment:
looks like unreachable code?

Reply:
no, the first 2 statements do not do return